### DISCUSSION TITLE
Move commits up and down within TMUX

### DISCUSSION THREAD

**Discussion by baggiponte** [jesseduffield/lazygit#3201] [Q&A]
Hi! I use a lot lazygit to rearrange commits (dope feature) with ctrl-j/k. I recently switched to tmux and I use ctrl-h/j/k/l to move around panes (through vim-tmux-navigator). The result is I cannot currently rearrange commits. What shortcuts would you replace them with? Thanks!

**Comment by mark2185** [comment]
I have the same "problem", but instead of remapping I just run `<tmux-prefix>:send-keys c-j` and it works. It's easier than opening up the menu with `?` and selecting `Move commit down/up one`.

**Reply by baggiponte** [reply]
Thanks for the reply! How/where do you configure `<tmux-prefix>:send-keys c-j`?

**Reply by mark2185** [reply]
I don't, actually, I type it once and then use the history feature to run it again/change it to the other direction.

To be clear, it's just a `tmux` command that sends those keys to the underlying pane, if you wish to bind it to something, you'll have to do it in `.tmux.conf`.

**Reply by baggiponte** [reply]
How does it not create a conflict with `c-j` being a way to swap within panes in tmux?

**Reply by mark2185** [reply]
Depends on how the keys are mapped in the first place, e.g. I have this in my `.tmux.conf`:
```
bind -n C-j if "$is_vim" "send-keys C-j"  "select-pane -D"
bind -n C-k if "$is_vim" "send-keys C-k"  "select-pane -U"
```

The `bind` isn't recursive so explicitly sending `C-j` doesn't go through the same process.

**Reply by baggiponte** [reply]
Oh I did not know about `if`! How can I define a `$is_lazygit`?

**Reply by mark2185** [reply]
This seems to work

```
# use c-k/c-j to move commits up/down within lazygit
is_lg="ps -o state= -o comm= -t '#{pane_tty}' | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?lazygit$'"

bind -n C-j if "$is_lg" "send-keys C-j"  "select-pane -D"
bind -n C-k if "$is_lg" "send-keys C-k"  "select-pane -U"
```

**Reply by baggiponte** [reply]
Thank you!

**Reply by asmmartin** [reply]
For me, that made C-j and C-k work in lazygit, but stopped working for moving between vim panels. So, I placed both checks:

```
is_vim="ps -o state= -o comm= -t '#{pane_tty}' | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?(view|l?n?vim?x?|fzf)(diff)?$'"
is_lg="ps -o state= -o comm= -t '#{pane_tty}' | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?lazygit$'"
bind -n C-j if "$is_lg || $is_vim" "send-keys C-j"  "select-pane -D"
bind -n C-k if "$is_lg || $is_vim" "send-keys C-k"  "select-pane -U"
```

**Reply by baggiponte** [reply]
That's clever too! In the end, I remapped the "move commit" up and down to capital J and K (i.e. shift-k,j).

**Comment by mloskot** [comment]
This a bit simpler variation should work too (works for me):

```tmux
bind -n C-j if-shell "[ #{pane_current_command} = lazygit ]" "send-keys C-j"  "select-pane -D"
bind -n C-k if-shell "[ #{pane_current_command} = lazygit ]" "send-keys C-k"  "select-pane -U"
```

**Comment by KennethHoff** [comment]
I had no idea you could rearrange commits because I have only ever used lg in tmux with vim-tmux-navigator 🫠
